### PR TITLE
BUGFIX: don't crash Meson on bad existing exe file: Fortran sanity check

### PR DESCRIPTION
**Pull request by scivision**
This fixes a not uncommon corner case for contemporary Windows 10 Fortran users that also use Windows Subsystem for Linux (WSL).

Replicate error:
1. build a Meson Fortran project in WSL
2. build the same project from native Windows without deleting the "build/meson-private" directory

Result: 
Meson hard crashes with OSError at mesonbuild/compilers/fortran.py:sanity_check at `Popen` for running the test executable, because it does not add the .exe extension, and Windows thinks you want to run `/meson-private/foo` instead of `meson-private/foo.exe`

Rather than monkey-patch the .exe extension onto the test executable filename, it seemed better to have Meson cleanup the conflicting WSL executable and proceed to build--at least not crashing Meson altogether.

### CODE REVIEW

File: mesonbuild/compilers/fortran.py [hunk 1]
```diff
@@ -76,31 +76,44 @@ def get_no_warn_args(self):
...
-        pe.wait()
-        if pe.returncode != 0:
+            cmdlist = [str(binary_name)]
+        # %% Run the test executable
+        try:
+            pe = subprocess.Popen(cmdlist, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+            pe.wait()
+            if pe.returncode != 0:
+                raise EnvironmentException('Executables created by Fortran compiler %s are not runnable.' % self.name_string())
+        except OSError:
```
Comment (dcbaker) [inline mesonbuild/compilers/fortran.py]: We can simplify this by using one of the subprocess helpers:
```python
try:
    subprocess.run(cmdline, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, check=True)
except (CalledProcessError, OSError):  # or maybe (PermisionDenied, FileNotFound)?
    raise EnvironmentError(...)
```

File: mesonbuild/compilers/fortran.py [hunk 2]
```diff
@@ -76,31 +76,44 @@ def get_no_warn_args(self):
...
+
         if environment.is_cross_build() and not self.is_cross:
             for_machine = MachineChoice.BUILD
         else:
             for_machine = MachineChoice.HOST
         extra_flags = environment.coredata.get_external_args(for_machine, self.language)
         extra_flags += environment.coredata.get_external_link_args(for_machine, self.language)
-        pc = subprocess.Popen(self.exelist + extra_flags + [source_name, '-o', binary_name])
+        # %% build the test executable
+        pc = subprocess.Popen(self.exelist + extra_flags + [str(source_name), '-o', str(binary_name)])
```
Comment (dcbaker) [inline mesonbuild/compilers/fortran.py]: see my next comment here too.

**Comment by dcbaker** [comment]
lol, never mind.

**Comment by scivision** [comment]
Yea I wasn't sure; was thinking of a time where asyncio might be used in general for Meson and then they'd all be Popen again via asyncio subprocess.

**Comment by dcbaker** [comment]
I doubt that meson will ever go full async, but I've been surprised before :)

**Pull request merged**